lib/mysql_connection.cpp

```diff
@@ -119,18 +119,18 @@ void MySQL_Connection_userinfo::set(MySQL_Connection_userinfo *ui) {
 
 
 bool MySQL_Connection_userinfo::set_schemaname(char *_new, int l) {
-	int _l=0;
-	if (schemaname) {
-		_l=strlen(schemaname); // bug fix for #609
-	}
-	if ((schemaname==NULL) || (strncmp(_new,schemaname, (l > _l ? l : _l) ))) {
```

Comment:
That seems a little magic to me. Wouldn't comparing "clean" memory be easier and safer than relying on signals like string length?